### PR TITLE
fix(format): skip `avio_open` when format has `AVFMT_NOFILE`

### PR DESCRIPTION
**Pull request by encounter**
Avoids creating empty files for parameterized output paths and similar. (e.g. `output_%v.m3u8` for HLS)

### CODE REVIEW

File: src/format/mod.rs [hunk 1]
```diff
@@ -191,12 +209,16 @@ pub fn output_with(path_or_url: impl AsRef<OsStr>, options: Dictionary) -> Resul
 
 		match avformat_alloc_output_context2(&mut ps, ptr::null(), ptr::null(), path.as_ptr()) {
 			0 => {
+				let output = context::Output::wrap(ps);
+				if (output.format().flags() & Flags::NO_FILE) == Flags::NO_FILE {
+					return Ok(output);
```
Comment (tilpner) [inline src/format/mod.rs:214]: This would skip the `Dictionary::own` below, causing a memory leak.

File: src/format/mod.rs [hunk 2]
```diff
@@ -54,10 +54,16 @@ pub fn open(path_or_url: impl AsRef<OsStr>, format: &Format) -> Result<Context,
 
 			Format::Output(ref format) => {
 				match avformat_alloc_output_context2(&mut ps, format.as_ptr() as *mut _, ptr::null(), path.as_ptr()) {
-					0 => match avio_open(&mut (*ps).pb, path.as_ptr(), AVIO_FLAG_WRITE) {
-						0 => Ok(Context::Output(context::Output::wrap(ps))),
-						e => Err(Error::from(e)),
-					},
+					0 => {
+						let output = context::Output::wrap(ps);
+						if (output.format().flags() & Flags::NO_FILE) == Flags::NO_FILE {
```
Comment (tilpner) [inline src/format/mod.rs:59]: These can be `output.format().flags().contains(Flags::NO_FILE)`.

**Review by tilpner — CHANGES_REQUESTED** [review]

**Comment by tilpner** [comment]
Good catch, thanks for the PR :)

**Comment by encounter** [comment]
Adjusted to resolve the dictionary leaks, thanks! I'm going to submit a follow-up PR to clean up the Dictionary disown/own uses further.

**Pull request merged**